src/libexpr/eval.cc

```diff
@@ -1348,7 +1348,7 @@ void EvalState::autoCallFunction(Bindings & args, Value & fun, Value & res)
     }
 
     Value * actualArgs = allocValue();
-    mkAttrs(*actualArgs, fun.lambda.fun->formals->formals.size());
+    mkAttrs(*actualArgs, std::max((uint32_t) fun.lambda.fun->formals->formals.size(), args.size()));
```

Comment:
generally I'd push for a `static_cast` here over a C-style cast.

Reply:
Thanks, did that, I literally have no clue about C++ :)